pennylane/compiler/__init__.py

```diff
@@ -0,0 +1,221 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+.. currentmodule:: pennylane
+
+This module provides support for hybrid quantum-classical compilation.
+Through the use of the :func:`~.qjit` decorator, entire workflows
+can be just-in-time (JIT) compiled --- including both quantum and
+classical processing --- down to a machine binary on first
+function execution. Subsequent calls to the compiled function will execute
+the previously compiled binary, resulting in significant
+performance improvements.
+
+Currently, PennyLane supports the
+`Catalyst <https://github.com/pennylaneai/catalyst>`__ hybrid compiler
+with the :func:`~.qjit` decorator. A significant benefit of Catalyst
+is the ability to preserve complex control flow around quantum
+operations — such as if statements and for loops, and including measurement
+feedback — during compilation, while continuing to support end-to-end
+autodifferentiation.
+
+.. note::
+
+    Catalyst currently only supports the JAX interface of PennyLane.
+
+Overview
+--------
+
+The main entry point to hybrid compilation in PennyLane
+is via the qjit decorator.
+
+.. autosummary::
+    :toctree: api
+
+    ~qjit
+
+In addition, several developer functions are available to probe
+available hybrid compilers.
+
+.. autosummary::
+    :toctree: api
+
+    ~compiler.available_compilers
+    ~compiler.available
+    ~compiler.active_compiler
+    ~compiler.active
+
+Compiler
+--------
+
+The compiler module provides the infrastructure to integrate external
+hybrid quantum-classical compilers with PennyLane, but does not provide
+a built-in compiler.
+
+Currently, only the `Catalyst <https://github.com/pennylaneai/catalyst>`__
+hybrid compiler is supported with PennyLane, however there are plans
+to incorporate additional compilers in the near future.
+
+.. note::
+
+    Catalyst is officially supported on Linux (x86_64) and macOS (aarch64) platforms.
+    To install it, simply run the following ``pip`` command:
+
+    .. code-block:: console
+
+      pip install pennylane-catalyst
+
+    Please see the :doc:`installation <catalyst:dev/installation>`
+    guide for more information.
+
+Basic usage
+-----------
+
+.. note::
+
+    Catalyst supports compiling QNodes that use ``lightning.qubit``,
+    ``lightning.kokkos``, ``braket.local.qubit``, and ``braket.aws.qubit``
+    devices. It does not support ``default.qubit``.
+
+    Please see the :doc:`Catalyst documentation <catalyst:index>` for more details on supported
+    devices, operations, and measurements.
+
+When using just-in-time (JIT) compilation, the compilation is triggered at the call site the
+first time the quantum function is executed. For example, ``circuit`` is
+compiled as early as the first call.
```

Comment:
```suggestion
compiled in the first call.
```
Not sure if true, but the ambiguity confused me.

Reply:
Fixed  #4812